File: frontend/src/app/features/dashboard/assignments/dtos/assignment-response.dto.ts
import {BaseResponseDto} from '../../../../core/dtos/base-response.dto';
import {KeyResponseDto} from '../../keys/dtos/key-response.dto';
import {ApplicantResponseDto} from '../../applicants/dtos/applicant.response.dto';

export interface AssignmentResponseDto extends BaseResponseDto {
  startTime: string;
  endTime: string;
  date: Date;
  key: KeyResponseDto
  applicant: ApplicantResponseDto;
}

File: backend/src/main/java/io/hb3nce04/keykeeperapp/feature/assignment/model/dto/response/AssignmentResponseDto.java
package io.hb3nce04.keykeeperapp.feature.assignment.model.dto.response;

import java.time.LocalDate;
import java.time.LocalTime;

import com.fasterxml.jackson.annotation.JsonFormat;
import io.hb3nce04.keykeeperapp.feature.key.model.dto.response.KeyResponseDto;
import io.hb3nce04.keykeeperapp.feature.applicant.model.dto.ApplicantResponseDto;
import io.hb3nce04.keykeeperapp.common.model.dto.BaseDto;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class AssignmentResponseDto extends BaseDto {
    @JsonFormat(pattern = "HH:mm:ss")
    private LocalTime startTime;

    @JsonFormat(pattern = "HH:mm:ss")
    private LocalTime endTime;

    private LocalDate date;

    private KeyResponseDto key;

    private ApplicantResponseDto applicant;
}
